### PR TITLE
feat(ONB-201): Keyword Generator Commit

### PR DESCRIPTION
**Pull request by eececan**
Keyword Generator application is completed.

### CODE REVIEW

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 1]
```diff
@@ -1,14 +1,11 @@
 <script setup>
-import { ref } from 'vue';
+import { ref,computed } from 'vue';
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:2]:
```suggestion
import { ref, computed } from 'vue';
```

Comment (tinas) [reply src/pages/keyword-generator/KeywordGenerator.vue:2]: In order to increase the readability of the code, we must take care of its formatting.

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
 
 const text = ref('');
-const onegrams = ref([]);
-const twograms = ref([]);
-const threegrams = ref([]);
 const keywords = ref([]);
-const showWords = ref(false);
+const maxNGrams = ref(10);
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:6]: Why `maxNGrams` is 10?

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 3]
```diff
@@ -18,62 +15,45 @@ const generate = (keywords, n) => {
...
-    threegrams.value = generate(text.value, 3);
-    keywords.value = [...onegrams.value, ...twograms.value, ...threegrams.value];
-    showWords.value = true;
+const generateNGrams = () => {
+  keywords.value = [];
+  if (text.value.trim() !== '') {
+    for(let i = 1;i <= maxNGrams.value; i++) {
+      keywords.value.push({i,keywords: generateWordArray(text.value, i)});
+    }
   }
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:24]: Here the code only continues if the condition is met. So we can use early return if the condition is not met.

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 4]
```diff
@@ -18,62 +15,45 @@ const generate = (keywords, n) => {
...
 }
+const validNGrams = computed(()=> {
+  return keywords.value.filter(nGram=>nGram.keywords.length > 0);
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:27]:
```suggestion
  return keywords.value.filter(nGram => nGram.keywords.length);
```

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 5]
```diff
@@ -18,62 +15,45 @@ const generate = (keywords, n) => {
...
-          <li v-for="(keyword, index) in onegrams" :key="'onegram-' + index">{{ keyword }}</li>
-        </ul>
-      </div>
-      <div v-if="twograms.length > 0">
-        <h4>Two-gram keywords:</h4>
+      <div v-for="nGram in validNGrams" :key="nGram.i">
+        <h4>{{ nGram.i }}-gram keywords:</h4>
         <ul>
-          <li v-for="(keyword, index) in twograms" :key="'twogram-' + index">{{ keyword }}</li>
+          <li v-for="(keyword,index) in nGram.keywords" :key="nGram.i + index">{{keyword}}</li>
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:49]:
```suggestion
          <li v-for="keyword in nGram.keywords" :key="nGram.i + index">{{keyword}}</li>
```

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 6]
```diff
@@ -18,62 +15,45 @@ const generate = (keywords, n) => {
...
     <div class="ma-header">
       <h3>Keyword Generator</h3>
     </div>
 
     <div class="ma-body">
-      <textarea v-model="text" placeholder="Enter a text here..."></textarea>
-      <button @click="run">Generate</button>
+      <div class="text-input">
+        <textarea v-model="text" placeholder="Enter a text here..."></textarea>
+      </div>
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:40]:
```suggestion
      <textarea v-model="text" placeholder="Enter a text here..." class="text-input"></textarea>
```

**Review by tinas — CHANGES_REQUESTED** [review]

File: src/pages/keyword-generator/KeywordGenerator.vue (hunk 2 above)
Comment (eececan) [reply src/pages/keyword-generator/KeywordGenerator.vue:6]: In the task description it was written that the application should show up to 10-gram keywords.

File: src/pages/keyword-generator/KeywordGenerator.vue (hunk 5 above)
Comment (eececan) [reply src/pages/keyword-generator/KeywordGenerator.vue:49]: When I do the suggested change I get the following error: 
" ESLint: Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive. (vue/valid-v-for) "

Comment (eececan) [reply src/pages/keyword-generator/KeywordGenerator.vue:49]: However, the application is working as it should. Is this a problem?

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 7]
```diff
@@ -1,9 +1,100 @@
 <script setup>
+import { ref, computed } from 'vue';
+
+const text = ref('');
+const keywords = ref([]);
+
+const generateWordArray = (keywords, n) => {
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:7]:
```suggestion
const generateNGram = (keywords, n) => {
```

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 8]
```diff
@@ -1,9 +1,100 @@
...
+    const nGram = words.slice(i, i + n).join(' ');
+    result.add(nGram);
+  }
+  return Array.from(result);
+}
+
+const generateNGrams = () => {
+  keywords.value = [];
+  if (!text.value.trim()) { return; }
+  for(let i = 1;i <= 3; i++) {
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:20]:
```suggestion
  for(let i = 1; i <= 3; i++) {
```

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 9]
```diff
@@ -1,9 +1,100 @@
...
+    keywords.value.push({i,keywords: generateWordArray(text.value, i)});
+  }
+}
+const validNGrams = computed(()=> {
+  return keywords.value.filter(nGram => nGram.keywords.length);
+});
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:26]: We can pass this computed property directly to the `generateNGrams` function. No need to filter it afterwards.

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 10]
```diff
@@ -1,9 +1,100 @@
...
+  font-family: Arial;
+  background-color: #cae3ff;
+  width: 100%;
+  height: 100px;
+  margin-bottom: 10px;
+  border-style: groove;
+  border-radius: 4px;
+}
+
+button {
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:86]: We should not style the html element directly.

**Review by tinas — CHANGES_REQUESTED** [review]

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 11]
```diff
@@ -1,9 +1,99 @@
...
+        <h4>{{ nGram.i }}-gram keywords:</h4>
+        <ul>
+          <li v-for="keyword in nGram.keywords" :key="nGram.i + index">{{keyword}}</li>
+        </ul>
+      </div>
+    </div>
+
+  </div>
+</template>
+<style scoped>
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue:51]: To keep the readability of the code at a high level, we must pay attention to the code layout. We should avoid extra spaces. We should always leave space to the left and right of the operators.

**Review by tinas — CHANGES_REQUESTED** [review]

File: src/pages/keyword-generator/KeywordGenerator.vue [hunk 12]
```diff
@@ -1,9 +1,95 @@
 <script setup>
+import { ref } from 'vue';
+const nGrams = ref([]);
```
Comment (tinas) [inline src/pages/keyword-generator/KeywordGenerator.vue]:
```suggestion
import { ref } from 'vue';

const nGrams = ref([]);
```

**Review by tinas — APPROVED** [review]

**Pull request merged**